libs/permit-utils/src/types.ts

```diff
@@ -2,26 +2,27 @@ import { Eip2612PermitUtils } from '@1inch/permit-signed-approvals-utils'
 import { latest } from '@cowprotocol/app-data'
 import { JsonRpcProvider } from '@ethersproject/providers'
 
-export type PermitType = 'dai-like' | 'eip-2612'
+export type PermitType = 'dai-like' | 'eip-2612' | 'unsupported'
 
-export type SupportedPermitInfo = {
+export type PermitInfo = {
   type: PermitType
-  version: string | undefined // Some tokens have it different than `1`, and won't work without it
+  // TODO: make it not optional once token-lists is migrated
+  name?: string
+  version?: string | undefined // Some tokens have it different than `1`, and won't work without it
 }
-type UnsupportedPermitInfo = false
-export type PermitInfo = SupportedPermitInfo | UnsupportedPermitInfo
```

Comment:
Biggest change:
1. Remove `false` version in favor of `unsupported` type
2. Add `name`